source/simulator/stokes_matrix_free.cc

```diff
@@ -1422,6 +1422,14 @@ namespace aspect
 
                       active_cell_data.newton_factor_wrt_pressure_table(cell,q)[i]
                         = derivatives->viscosity_derivative_wrt_pressure[q] * newton_derivative_scaling_factor;
+                      Assert(!std::isnan(active_cell_data.newton_factor_wrt_pressure_table(cell,q)[i]),
```

Comment:
how about using ``std::isfinite`` instead?

Reply:
ah, I wasn't aware that that included nan's. Thanks! I update it.